tests/base/__init__.py

```diff
@@ -43,6 +44,19 @@ def override_config(self, integration, values):
         finally:
             options.update(original)
 
+    @contextlib.contextmanager
```

Comment:
Isn't this a duplication of: https://github.com/DataDog/dd-trace-py/blob/da8a340d5c5e32cb92f31c93a389ff4204627710/tests/util.py#L57-L67 ?

Reply:
If we want to move an utility somewhere else (better organization etc...) you may:
* move the utility in this new module
* in the previous one (so `util.py`) just import this so it's still available
* in a separated PR migrate all the tests to use the new import so you can remove the import from `util.py`

Reply:
@palazzem yes, we have a small number of duplicated ones as we move test to use this new base test case class.

will keep an eye on this migration.